src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -5945,6 +5947,10 @@ void CodeGen::genCompareInt(GenTree* treeNode)
     }
     else if (op1->isUsedFromReg() && op2->IsIntegralConst(0))
     {
+        if ((treeNode->OperIs(GT_EQ) || treeNode->OperIs(GT_NE)) && compiler->opts.OptimizationEnabled())
```

Comment:
Currently, it will (nearly?) always be the case that `treeNode->gtPrev == op1` in this case. However, I think that you need to check for that here to ensure that other instructions that set flags have not be emitted after `op1`.

Reply:
I don't think we need the check since we're basing the logic off of the last emitted instruction.